client/lxd_images.go

```diff
@@ -608,6 +608,115 @@ func (r *ProtocolLXD) CopyImage(source ImageServer, image api.Image, args *Image
 		return nil, err
 	}
 
+	// Relay mode
+	if args != nil && args.Mode == "relay" {
+		metaFile, err := ioutil.TempFile("", "lxc_image_")
+		if err != nil {
+			return nil, err
+		}
+		defer os.Remove(metaFile.Name())
+
+		rootfsFile, err := ioutil.TempFile("", "lxc_image_")
+		if err != nil {
+			return nil, err
+		}
+		defer os.Remove(rootfsFile.Name())
+
+		// Import image
+		req := ImageFileRequest{
+			MetaFile:   metaFile,
+			RootfsFile: rootfsFile,
+		}
+
+		_, err = source.GetImageFile(image.Fingerprint, req)
+		if err != nil {
+			return nil, err
+		}
+
+		// Export image
+		_, err = metaFile.Seek(0, 0)
+		if err != nil {
+			return nil, err
+		}
+
+		_, err = rootfsFile.Seek(0, 0)
+		if err != nil {
+			return nil, err
+		}
+
+		imagePost := api.ImagesPost{}
+		imagePost.Public = args.Public
+
+		createArgs := &ImageCreateArgs{
+			MetaFile:   metaFile,
+			MetaName:   image.Filename,
+			RootfsFile: rootfsFile,
+			RootfsName: image.Filename,
+			Type:       image.Type,
+		}
+
+		rop := remoteOperation{
+			chDone: make(chan bool),
+		}
+
+		// For older servers, apply the aliases after copy
+		if !r.HasExtension("image_create_aliases") && image.Aliases != nil {
+			rop.chPost = make(chan bool)
+
+			go func() {
+				defer close(rop.chPost)
+
+				// Wait for the main operation to finish
+				<-rop.chDone
+				if rop.err != nil {
+					return
+				}
```

Comment:
Isn't this racy?

Once chDone is closed, there's nothing preventing the CLI from just exiting before this goroutine ever has a chance to do anything.

Shouldn't this logic be a defer of the other goroutine instead?

Reply:
I copied this code from the `tryCopyImage` function and just modified it a bit (single server instead of multiple servers). This should probably be fine then, no?

Reply:
So actually, why do we have any of that code?

Push support will only work on new servers, so we shouldn't need to care about any of the late Alias apply stuff.